x-pack/plugins/monitoring/public/application/pages/use_setup_mode_supported.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+
+// List of RegEx for the routes that use <SetupModeRenderer>
+const pagesWithSetupModeSupported = [
```

Comment:
Is there a way to have the SetupMode wrapper be the authority about whether it is supported or not by a page so we don't have to define an allowlist that has to be maintained ? Including the SetupMode in a page is the only thing I need to enable it

I'm also wondering if this list is currently too permissive, for example `ml_jobs` does not use the [setupMode injected property](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/public/application/pages/elasticsearch/ml_jobs_page.tsx#L95) which seems to do the heavy lifting, and I don't know what we could setup there ?

Reply:
Ah, we need to have this additional logic because the buttons are now part of the navigation bar which we don't have control over in the wrapper

Reply:
I like that idea more, let me try to craft something like that!